validmind/vm_models/dataset.py

```diff
@@ -448,20 +525,67 @@ def assign_predictions(  # noqa: C901 - we need to simplify this method
             self._extra_columns.setdefault("prediction_columns", {})[
                 model.input_id
             ] = prediction_column
+
+        # Step 4: Prediction Values Provided without Specific Column
         elif prediction_values is not None:
             if len(prediction_values) != self.df.shape[0]:
                 raise ValueError(
                     "Length of prediction values doesn't match number of rows of the dataset"
                 )
             pred_column = f"{model.input_id}_prediction"
             if pred_column in self.columns:
-                raise ValueError(
-                    f"Prediction column {pred_column} already exists in the dataset"
+                warnings.warn(
+                    f"Prediction column {pred_column} already exists in the dataset, overwriting the existing predictions",
+                    UserWarning,
```

Comment:
Rather than overwriting the existing predictions, it would be nice to pass `overwrite_flag = True` as a method argument. It might happen that user forget to change `input_id` of the model.